crates/biome_js_analyze/src/lint/nursery/no_document_cookie.rs

```diff
@@ -0,0 +1,111 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_js_syntax::{
+    global_identifier, AnyJsAssignment, AnyJsAssignmentPattern, JsAssignmentExpression,
+};
+use biome_rowan::AstNode;
+
+use crate::services::semantic::Semantic;
+
+declare_lint_rule! {
+    /// Disallow use `document.cookie` directly.
+    ///
+    /// It's not recommended to use document.cookie directly as it's easy to get the string wrong.
+    /// Instead, you should use the [Cookie Store API](https://developer.mozilla.org/en-US/docs/Web/API/CookieStore) or a [cookie library](https://www.npmjs.com/search?q=cookie).
```

Comment:
I don't think we should link to npm search queries. We should make specific recommendations, or not make them at all.